cql3/values.cc

```diff
@@ -58,4 +58,22 @@ raw_value_view::raw_value_view(managed_bytes&& tmp) {
     _data = managed_bytes_view(*_temporary_storage);
 }
 
+bool operator==(const raw_value& v1, const raw_value& v2) {
+    return std::visit(overloaded_functor{
+        [] <typename T> (const T& s1, const T& s2) {
+            return s1 == s2;
```

Comment:
Do we want to say that `NULL == NULL` is true? Usually in CQL `NULL` cannot equal anything, even another `NULL`.
`raw_value` is often used to evaluate conditions, for example in `is_satisfied_by`. Currently it's handled correctly because it doesn't use this `operator==`, but I wonder if sometime in the future someone isn't going to use `operator==` by mistake thinking it handles `NULL` in the cql way.

Maybe instead of `operator==` we could have two functions `equal_cql` and `equal_data` or something? Just thinking out loud.

Reply:
This is not CQL equals, it's C++ object equals (later used to implement expression equals).

CQL "NULL = NULL" is CQL FALSE

But (CQL "NULL = NULL") == (CQL "NULL = NULL") is C++ true - they are both the same expressions.

(Note: (CQL "NULL = NULL") == (CQL "NULL") can be evaluated as false by a simple implementation, or true by a sophisticated implementation, and both results are correct)